notebooks/DATB/psf_grid_simulation_photometry/GriddedPSFModel_simulation_and_photometry.ipynb

```diff
@@ -0,0 +1,471 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We start by loading into FITS HDU a pre-generated grid of JWST PSFs for NIRCam detector A1 in F090W."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "url = 'https://stsci.box.com/shared/static/6h8wsr2ts0t24s79cxmnyk8bldt0vv3i.fits'\n",
+    "hdu = fits.open(url)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We now create a `photutils.GriddedPSFModel` from the FITS HDU."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "psfmodel = to_griddedpsfmodel(hdu, ext=0)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "psfmodel"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "`psfmodel` is an instance of a `photutils.GriddedPSFModel`.  Let's view some of it's attributes."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The oversampled PSF grid is stored internally as a 3D `numpy.ndarray`.  Here there are 16 PSFs (from a 4x4 reference grid) and the shape of each is 404x404 pixels."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "psfmodel.data.shape"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The `oversampling` attribute is the PSF oversampling factor."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "psfmodel.oversampling"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The `grid_xypos` attribute contains a list of the (`x, y`) positions of reference PSFs."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "psfmodel.grid_xypos"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This model contains a 4x4 grid of reference PSFs."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "len(psfmodel.grid_xypos)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The `meta` attribute is dictionary holding detailed information how the PSF model grid was generated by `webbpsf`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "psfmodel.meta"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can use the `webbpsf.gridded_library.display_psf_grid` function to visualize the PSF grid."
```

Comment:
An aside thought: maybe thise should be moved to photutils at some point?  Since it's probably useful more generically for `GriddedPSFModel`.  And if so, might want to mention that so that a future reader of this nb knows to look for an alternate location.

Reply:
Yes, good idea.